src/main/java/net/openhft/chronicle/queue/ExcerptTailer.java

```diff
@@ -290,4 +319,13 @@ default long excerptsInCycle(int cycle) {
     default @NotNull ExcerptTailer afterWrittenMessageAtIndex(@NotNull ChronicleQueue queue, long index) {
         throw new UnsupportedOperationException("todo");
     }
+
+    interface AcknowledgedIndexReplicatedCheck {
+        /**
+         * @param index           the index of the next message for the tailer to read
```

Comment:
I would recommend adding a Javadoc here letting implementers know they need to be careful around the roll when calculating delta between index values and should lean on cycle calculations as needed if correctness in terms of index delta calculation is critical for them during the roll